src/Avalonia.Controls/Primitives/Popup.cs

```diff
@@ -586,6 +586,19 @@ private void CloseCore()
             }
 
             Closed?.Invoke(this, EventArgs.Empty);
+
+            if(PlacementTarget != null)
+            {
+                FocusManager.Instance?.Focus(PlacementTarget);
+            }
+            else
+            {
+                var anc = this.GetLogicalAncestors().OfType<IControl>().FirstOrDefault();
```

Comment:
```suggestion
                var anc = this.FindLogicalAncestorOfType<IControl>();
```

Reply:
Fixed. I also fixed this in `Open()` where I originially copied this from